src/sqlfluff/dialects/dialect_spark3.py

```diff
@@ -783,3 +783,118 @@ class JoinClauseSegment(BaseSegment):
     get_eventual_alias = ansi_dialect.get_segment(
         "JoinClauseSegment"
     ).get_eventual_alias
+
+
+@spark3_dialect.segment(replace=True)
+class TableExpressionSegment(BaseSegment):
+    """The main table expression e.g. within a FROM clause."""
+
+    type = "table_expression"
+    match_grammar = OneOf(
+        Ref("ValuesClauseSegment"),
+        Ref("BareFunctionSegment"),
+        Ref("FunctionSegment"),
+        Ref("TableReferenceSegment"),
+        # Nested Selects
+        Bracketed(Ref("SelectableGrammar")),
+    )
+
+
+@spark3_dialect.segment(replace=True)
+class AliasExpressionSegment(BaseSegment):
+    """A reference to an object with an `AS` clause.
+
+    The optional AS keyword allows both implicit and explicit aliasing.
+    Note also that it's possible to specify just column aliases without aliasing the table as well:
+    .. code-block:: sql
+
+        SELECT * FROM VALUES (1,2) as t (a, b);
+        SELECT * FROM VALUES (1,2) as t;
+        SELECT * FROM VALUES (1,2) as (a, b);
+
+    """
+
+    type = "alias_expression"
+    match_grammar = Sequence(
+        Ref.keyword("AS", optional=True),
+        OneOf(
+            Sequence(
+                # maybe table alias
+                Ref("SingleIdentifierGrammar", optional=True),
+                # maybe column aliases
+                Bracketed(Ref("SingleIdentifierListSegment"), optional=True),
+            ),
+            Ref("QuotedLiteralSegment"),
+        ),
```

Comment:
```suggestion
        OneOf(
            Sequence(
                # maybe table alias
                Ref("SingleIdentifierGrammar", optional=True),
                # maybe column aliases
                Bracketed(Ref("SingleIdentifierListSegment")),
            ),
            # maybe table alias
            Ref("SingleIdentifierGrammar"),
            Ref("QuotedLiteralSegment"),
        ),
```
Not sure if there's a risk of something weird happening if both the table and column aliases are optional so maybe best to be more explicit about the options.

Also what is the situation in which we use `QuotedLiteralSegment`?
Not too familiar with spark3 myself but `SELECT * FROM VALUES (1,2) as 'I am just a string';` seems a bit odd. What does the string represent?

Reply:
The `QuotedLiteralSegment` was copy-pasta  from [/src/sqlfluff/dialects/dialect_ansi.py#L820](https://github.com/mcannamela/sqlfluff/blob/84e61e00f6cfcbd8cd5b7e3f3ca100885f5c12f6/src/sqlfluff/dialects/dialect_ansi.py#L820). It struck me as odd as well. 

I'll play around with it in the spark dialect